### ISSUE TITLE
Use proper intra doc links for macro generated code

### ISSUE THREAD

**Issue by thiolliere** [paritytech/substrate#9962]
doc like https://github.com/paritytech/substrate/blob/62f011ca8d0491f6be77b3dbe0074eb060245d08/frame/support/procedural/src/construct_runtime/expand/origin.rs#L160 doesn't link to the item.

We should make a correct item link.

**Comment by zh-beep** [comment]
I'm currently learning Rust and would like to contribute, would I be able to help with this issue?

**Comment by dharjeezy** [comment]
Is this issue done? @thiolliere can I pick the issue up?

**Comment by thiolliere** [comment]
anybody can submit a PR which solve this issue, they can also ping my name on the PR top message.

On master I still see those which could have a intra doc link:
```
frame/support/procedural/src/construct_runtime/expand/origin.rs:185:			/// Create with system none origin and `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:189:			/// Create with system root origin and `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:193:			/// Create with system signed origin and `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:219:			/// Convert to runtime origin, using as filter: `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:251:			/// `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:314:			/// Convert to runtime origin using `frame-system::Config::BaseCallFilter`.
frame/support/procedural/src/construct_runtime/expand/origin.rs:78:		/// Origin is always created with the base filter configured in `frame_system::Config::BaseCallFilter`.
```

**Comment by dharjeezy** [comment]
can you check the PR i created @thiolliere

**Comment by dharjeezy** [comment]
@thiolliere i have updated the PR i opened. could you take a look now.

**Comment by KiChjang** [comment]
Fixed by #10271.
